### ISSUE TITLE
Make and deploy release on Bintray

### ISSUE THREAD

**Issue by juherr** [testng-team/testng-remote#1]
I made an organization on Bintray: https://bintray.com/testng-team

@missedone I can add you that will allow you to make a release there.

We will see later with @cbeust if/how to link it with central.

**Comment by missedone** [comment]
please let me know what can i do.
do we need authorize any script to build and upload the binaries to Bintray?

thanks

**Comment by cbeust** [comment]
I think @juherr looked into it and it doesn't seem possible for all members of the team to deploy, only one person. Was that right, @juherr?

**Comment by juherr** [comment]
It should be possible for the bintray team I created. I will try it.

Le lun. 11 janv. 2016 22:25, Cedric Beust <notifications@github.com> a
écrit :

**Comment by cbeust** [comment]
Also, I suspect one of us could create a key and then we all share it.

And uploading stuff to bintray is trivial with Kobalt ;-)

-- 
Cédric

**Comment by juherr** [comment]
In fact, it is not possible to share a personal bintray repo between 2 accounts, but it is possible with group repositories (what I did).
After, we will have to use your personal sonatype account to push from bintray to maven centrale (via bintray ui)

@missedone Do you have a bintray account?

**Comment by missedone** [comment]
just create one account: `missedone`
thanks

**Comment by missedone** [comment]
hi @juherr , @cbeust 
i just had a try with releasing onto bintary, it works as you can see artifacts uploaded here: https://bintray.com/testng-team/test/testng-remote/view#files/org/testng

i will create a DEPLOY.md later to document the release process,

**Comment by missedone** [comment]
as you might know, Bintary does not support uploading snapshot artifacts. only release bits can.

**Comment by missedone** [comment]
here is the release procedure: https://github.com/testng-team/testng-remote/blob/release_to_bintray/RELEASE.md

**Comment by juherr** [comment]
For snapshots, we can use [Sonatype](http://central.sonatype.org/pages/ossrh-guide.html) or [JFrog](https://www.jfrog.com/confluence/display/RTF/Deploying+Snapshots+to+oss.jfrog.org)

**Comment by missedone** [comment]
i created a new repo with package: https://bintray.com/testng-team/testng-extras/testng-remote/view to replace https://bintray.com/testng-team/test/testng-remote/view, what do you think about the name?

**Comment by juherr** [comment]
Or just https://bintray.com/testng-team/testng/testng-remote/view

**Comment by missedone** [comment]
im ok with this :+1:

**Comment by missedone** [comment]
later, we might need add travis-ci as well?

**Comment by juherr** [comment]
> later, we might need add travis-ci as well?

Yes! #20

**Comment by missedone** [comment]
it requires a sonatype account to sync jcenter to maven central:
<img width="1012" alt="maven_central" src="https://cloud.githubusercontent.com/assets/555134/16642495/713355a6-43c0-11e6-9673-e79fbaa6b0b1.png">
is there any existing account we can use?

**Comment by juherr** [comment]
Not yet, you can ask for a new one.

**Comment by juherr** [comment]
Rultor could be used to deploy on maven central if needed: http://www.yegor256.com/2014/08/19/how-to-release-to-maven-central.html

**Comment by juherr** [comment]
@missedone About your old question, I think @cbeust will have to set its credentials if we want to sync bintray with maven central because he owns the `org.testng` groupId

**Comment by juherr** [comment]
Up.

Bintray is dead and the artifact could not be used by others. For example, Gradle: see https://github.com/gradle/gradle/pull/19960

@krmahadevan I think you've already published a lot of artifacts and know what should be done, right?

**Comment by missedone** [comment]
@juherr @krmahadevan not sure who can set the gpg key on this github repo secrets for the maven publish.
also, it need @cbeust help to claim the ownership of the groupId org.testng, i'm not quite family with this process.

**Comment by juherr** [comment]
@krmahadevan and I can publish under the org.testng groupId.

@krmahadevan and you should be able to declare secrets under the testng-team github org. If not, ping me and I will promote your rights.

**Comment by krmahadevan** [comment]
> Up.
> 
> Bintray is dead and the artifact could not be used by others. For example, Gradle: see [gradle/gradle#19960](https://github.com/gradle/gradle/pull/19960)
> 
> @krmahadevan I think you've already published a lot of artifacts and know what should be done, right?

Yes. Here are the steps. First we need to add below secrets into the repository settings (I have access to that web page, but I dont know the actual secrete values there)

Below is a screenshot of the secrets that I have for my github organisation https:// github.com/rationaleemotions

![image](https://user-images.githubusercontent.com/909773/156696580-44705970-4652-4f5c-9385-8123ab3d1d51.png)

Then we can create a github workflow that looks like [this](https://github.com/RationaleEmotions/autospawn/blob/master/.github/workflows/release.yml)

We then alter the pom file to look like [this](https://github.com/RationaleEmotions/autospawn/blob/master/pom.xml#L118-L187) essentially adding a release profile

That should be sufficient. 

We also need to be removing [this distribution management](https://github.com/testng-team/testng-remote/blob/b8fd4153f4979aa08b3c31fd9deba3b541ebf909/pom.xml#L199-L209) section from our pom file and also remove [the repositories](https://github.com/testng-team/testng-remote/blob/b8fd4153f4979aa08b3c31fd9deba3b541ebf909/pom.xml#L183-L197) tag as well.

**Comment by krmahadevan** [comment]
> @krmahadevan and I can publish under the org.testng groupId.

I didnt know this :)

> @krmahadevan and you should be able to declare secrets under the testng-team github org. If not, ping me and I will promote your rights.

LIke I mentioned earlier, I can access the page, but I don't know what values to put in there.

**Comment by missedone** [comment]
is there any place we can publish the snapshot?

**Comment by juherr** [comment]
@missedone yes, see https://central.sonatype.org/publish/publish-maven/#other-prerequisites and  https://central.sonatype.org/publish/publish-guide/#accessing-repositories
